modules/raft-client/src/main/java/org/apache/ignite/raft/client/service/impl/RaftGroupServiceImpl.java

```diff
@@ -155,16 +187,48 @@ public RaftGroupServiceImpl(
         return learners;
     }
 
-    /** {@inheritDoc} */
-    @Override public CompletableFuture<Void> refreshLeader() {
+    /**
+     * Sends a {@link GetLeaderRequest}.
+     *
+     * @param groupId Raft group id.
+     * @param factory Message factory.
+     * @param cluster Cluster service.
+     * @param peers List of all peers.
+     * @param timeout Timeout.
+     * @param retryDelay Retry delay.
+     * @return Future representing pending completion of the request.
+     */
+    private static CompletableFuture<Peer> getLeader(
```

Comment:
Can you make it not static and invoke it on new instance before completing the future in "start"? Too many arguments